src/commands/farm.rs

```diff
@@ -390,31 +398,19 @@ async fn process_block_stream(
     futures::pin_mut!(stream);
 
     let mut current_iter = 0;
+
     stream
-        // Map block number to block hash
-        .and_then(|block| {
-            match node.block_hash(block).into_eyre().transpose().map(futures::future::ready) {
-                Some(block_hash) => future::ok(Some(block_hash)),
-                None =>
-                    if !blocks_pruning {
-                        future::err(eyre!(
-                            "node database is probably corrupted, try wiping the node"
-                        ))
-                    } else {
-                        future::ok(None)
-                    },
-            }
+        .try_filter_map(|block| match node.block_hash(block).into_eyre() {
+            Ok(Some(block_hash)) => future::ok(Some(block_hash)),
+            Ok(None) if blocks_pruning => future::ok(None),
+            Ok(None) =>
+                future::err(eyre!("node database is probably corrupted, try wiping the node")),
+            // Err(err) => future::err(err.context("block hash couldn't found")),
+            Err(err) => future::err(eyre!("couldn't get block hash from node, because: {err}")),
```

Comment:
I think `Err(err) => future::err(err.wrap_err("block hash couldn't found"))` would work here.

It is really important to keep context everywhere. `eyre!` macro here will just expand latest error message, discarding any other ones